### PR TITLE
fix whereType case in rest adapters

### PR DESCRIPTION
**Pull request by mateominato**

### CODE REVIEW

File: packages/brick_offline_first_with_rest_build/lib/src/offline_first_rest_generators.dart [hunk 1]
```diff
@@ -134,18 +134,23 @@ class _OfflineFirstRestDeserialize extends RestDeserialize {
             final where =
                 _convertSqliteLookupToString(offlineFirstAnnotation.where!, iterableArgument: 's');
             final getAssociationText = getAssociationMethod(argType, query: 'Query(where: $where)');
-            final getAssociations = '''($fieldValue ?? []).map((s) => $getAssociationText
-            ).whereType<Future<$argType>>()$fromRestCast''';
 
             if (checker.isArgTypeAFuture) {
-              return getAssociations;
+              return '''($fieldValue ?? []).map<Future<$argType>>((s) => $getAssociationText)$fromRestCast''';
```
Comment (tshedor) [inline packages/brick_offline_first_with_rest_build/lib/src/offline_first_rest_generators.dart:139]: If this is one line it doesn't need to have a multiline string
```suggestion
              return '($fieldValue ?? []).map<Future<$argType>>((s) => $getAssociationText)$fromRestCast';
```

File: packages/brick_offline_first_with_rest_build/pubspec.yaml [hunk 2]
```diff
@@ -12,11 +12,14 @@ environment:
 dependencies:
   analyzer: ^1.2.0
   brick_offline_first_abstract: ^1.0.0+1
-  brick_build: ^1.2.2
+  brick_build:
```
Comment (tshedor) [inline packages/brick_offline_first_with_rest_build/pubspec.yaml:15]: The paths in all pubspecs should be reverted; instead it should be `brick_build: ^1.2.3` (I'll publish after this is merged)

**Review by tshedor — CHANGES_REQUESTED** [review]
Going to need CHANGELOG additions for `brick_offline_first_with_rest_build` and `brick_build`

File: packages/brick_offline_first_with_rest_build/pubspec.yaml (hunk 2 above)
Comment (mateominato) [reply packages/brick_offline_first_with_rest_build/pubspec.yaml:15]: i know, but i needed to do this for now in order to get a clean test run on CI

Comment (tshedor) [reply packages/brick_offline_first_with_rest_build/pubspec.yaml:15]: Understood; now that we have the green check mark on the latest commit we can revert these pubspecs

File: packages/brick_offline_first_with_rest_build/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
 ## Unreleased
 
+## 1.1.4
+
+* Always use `whereType<T>` casts after awaiting `Future.wait()` in adapters.
```
Comment (tshedor) [inline packages/brick_offline_first_with_rest_build/CHANGELOG.md:5]:
```suggestion
* Always use `whereType<T>` casts after awaiting `Future.wait()` in Rest deserializing adapters.
```

File: packages/brick_rest_generators/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 ## Unreleased
 
+## 1.0.3
+
```
Comment (tshedor) [inline packages/brick_rest_generators/CHANGELOG.md:4]:
```suggestion
```

File: packages/brick_rest_generators/pubspec.yaml [hunk 5]
```diff
@@ -4,15 +4,15 @@ homepage: https://github.com/greenbits/brick/tree/master/packages/brick_rest_gen
 issue_tracker: https://github.com/greenbits/brick/issues
 repository: https://github.com/greenbits/brick
 
-version: 1.0.2
+version: 1.0.3
```
Comment (tshedor) [inline packages/brick_rest_generators/pubspec.yaml:7]:
```suggestion
version: 1.0.2
```

File: packages/brick_sqlite_generators/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,9 @@
 ## Unreleased
 
+## 1.1.4
+
```
Comment (tshedor) [inline packages/brick_sqlite_generators/CHANGELOG.md:4]:
```suggestion
```

File: packages/brick_offline_first_with_rest_build/pubspec.yaml [hunk 7]
```diff
@@ -4,19 +4,19 @@ homepage: https://github.com/greenbits/brick/tree/master/packages/brick_offline_
...
   sdk: ">=2.12.0 <3.0.0"
 
 dependencies:
   analyzer: ^1.2.0
   brick_offline_first_abstract: ^1.0.0+1
-  brick_build: ^1.2.2
+  brick_build: ^1.2.3
   brick_rest: ^1.0.0+3
-  brick_rest_generators: ^1.0.2
+  brick_rest_generators: ^1.0.3
```
Comment (tshedor) [inline packages/brick_offline_first_with_rest_build/pubspec.yaml:17]:
```suggestion
  brick_rest_generators: ^1.0.2
```

File: packages/brick_offline_first_with_rest_build/pubspec.yaml [hunk 8]
```diff
@@ -4,19 +4,19 @@ homepage: https://github.com/greenbits/brick/tree/master/packages/brick_offline_
...
   brick_sqlite_abstract: ^1.0.0+1
-  brick_sqlite_generators: ^1.1.3
+  brick_sqlite_generators: ^1.1.4
```
Comment (tshedor) [inline packages/brick_offline_first_with_rest_build/pubspec.yaml:19]:
```suggestion
  brick_sqlite_generators: ^1.1.3
```

File: packages/brick_offline_first_with_rest_build/CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,12 @@
...
+* Bump `brick_build`
+* Bump `brick_sqlite_generators`
+* Bump `brick_rest_generators`
```
Comment (tshedor) [inline packages/brick_offline_first_with_rest_build/CHANGELOG.md:8]:
```suggestion
```

**Review by tshedor — CHANGES_REQUESTED** [review]
The REST and SQLite packages don't need to be bumped

**Pull request merged**